akka-http2-support/src/main/scala/akka/http/impl/engine/http2/FrameEvent.scala

```diff
@@ -4,37 +4,62 @@
 
 package akka.http.impl.engine.http2
 
-import akka.http.impl.engine.http2.Protocol.FrameType
-import akka.http.impl.engine.http2.Protocol.SettingIdentifier
+import akka.http.impl.engine.http2.Http2Protocol.FrameType
+import akka.http.impl.engine.http2.Http2Protocol.SettingIdentifier
 import akka.util.ByteString
 
-sealed trait FrameEvent
+sealed trait FrameEvent {
+  def flags: ByteFlag
+  def hasFlag(f: ByteFlag): Boolean = f.isSet(flags)
+}
 sealed trait StreamFrameEvent extends FrameEvent {
   def streamId: Int
 }
 
 final case class DataFrame(
-  streamId:  Int,
-  endStream: Boolean,
-  payload:   ByteString) extends StreamFrameEvent
+  flags:    ByteFlag,
+  streamId: Int,
+  payload:  ByteString) extends StreamFrameEvent {
+  def endStream = Http2Protocol.Flags.END_STREAM.isSet(flags)
+}
 
 final case class HeadersFrame(
+  flags:               ByteFlag,
```

Comment:
What do we win by removing the Boolean flags? 

Pro:
 * better packing, we might not waste a few bits per frame, leading to less GC in the best case

Cons:
 * have to move validation logic into the model classes ("which frames take which flags")
 * bit fiddling is moved from parsing / rendering into model
 * model case classes are less descriptive

So, for me it's not quite clear that this is a clear win (at this point, code clarity wins for me over potential performance optimizations).

How about this, we keep the `ByteFlag` to model the flags but still don't use it in the model classes but only in parser / renderer?